src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1254,21 +1254,26 @@ void ImageReplayer<I>::handle_process_entry_safe(const ReplayEntry &replay_entry
     m_remote_journaler->committed(replay_entry);
   }
 
-  m_perf_counters->inc(l_rbd_mirror_replay);
-  m_perf_counters->inc(l_rbd_mirror_replay_bytes,
-                       replay_entry.get_data().length());
-  m_perf_counters->tinc(l_rbd_mirror_replay_latency,
-                        ceph_clock_now() - replay_start_time);
+  auto bytes = replay_entry.get_data().length();
+  auto latency = ceph_clock_now() - replay_start_time;
+  auto ctx = new FunctionContext(
+    [this, bytes, latency](int r) {
+      if (g_perf_counters) {
```

Comment:
Shouldn't the global perf counter always be available? I'd vote to keep updating the global perf counters in-line and not move it to the lambda.

Reply:
I just wanted to keep this in one place. But ok, I will update it as you want.